### PR TITLE
Made the projects field optional #29

### PR DESCRIPTION
**Pull request by kiryteo**
The validation.js files from staticfiles and portfolio/static are changed.

### CODE REVIEW

**Comment by Anupam-dagar** [comment]
you edited regex method, firstname and password1 field.That doesn't need to be changed.
Also, corresponding changes needs to be done in display.html.If no projects are entered then there will be a weird blank space there.

File: portfolio/static/js/validation.js [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
   // Initialize form validation on the registration form.
   // It has the name attribute "registration"
-  $.validator.addMethod(
-        "regex",
-        function(value, element, regexp) {
-            var re = new RegExp(regexp);
-            return this.optional(element) || re.test(value);
-        },
-        "Please check your input."
-);
```
Comment (Anupam-dagar) [inline portfolio/static/js/validation.js]: Don't remove this

Comment (kiryteo) [reply portfolio/static/js/validation.js]: Sorry to edit the validation.js containing regex method, Now I've just edited the validation.js from staticfiles and display.html with required condition. Anything else required.?

Comment (Anupam-dagar) [reply portfolio/static/js/validation.js]: nope, commit your changes so that I can review

File: portfolio/templates/portfolio/display.html [hunk 2]
```diff
@@ -109,6 +109,7 @@ <h3>Skills</h3>
 				<h3>Projects</h3>
 				<p>Projects that I have worked on. </p>
 				<div class="features">
+					{% if detail.project1_name and detail.project1_description and detail.project1_url %}
```
Comment (Anupam-dagar) [inline portfolio/templates/portfolio/display.html]: It's fine but there should be something for the div with class features.If I submit without no projects then I will be able to see an empty block.

File: staticfiles/js/validation.js [hunk 3]
```diff
@@ -54,11 +54,11 @@ $(function() {
...
-      project1_name:"Please enter name of project1",
-      project1_url: {
-        required: "Please enter GitHub Link for the project",
-        url: "Please enter a valid url"
-      },
+   //   project1_name:"Please enter name of project1",
+   //   project1_url: {
+   //     required: "Please enter GitHub Link for the project",
+   //     url: "Please enter a valid url"
+   //   },
```
Comment (Anupam-dagar) [inline staticfiles/js/validation.js]: this file is not fixed.

**Comment by Anupam-dagar** [comment]
It seems like you are losing track of files.I would suggest you delete your current fork and start your work on a new fork.

**Comment by kiryteo** [comment]
All right, I'll work properly on a new fork.

**Pull request closed without merging**